### ISSUE TITLE
Crashes for apks built with minSdkVersion 24 or greater

### ISSUE THREAD

**Issue by grepx** [linkedin/dex-test-parser#12]
Hi,

I found this crash in the Flank test runner project, but it appears to be coming from the dependency with this project. I get the following crash on apks built with minSdkVersion 24 or greater:

```DexMagic(dex=[100, 101, 120], newline=10, version=[48, 51, 55], zero=0)
DexMagic(dex=[100, 101, 120], newline=10, version=[48, 51, 53], zero=0)
	at com.linkedin.dex.spec.DexMagic.validate(DexMagic.kt:31)
	at com.linkedin.dex.spec.HeaderItem.validate(HeaderItem.kt:62)
	at com.linkedin.dex.spec.DexFile.<init>(DexFile.kt:35)
	at com.linkedin.dex.parser.DexParser$Companion$readDexFiles$1$4.invoke(DexParser.kt:79)
	at com.linkedin.dex.parser.DexParser$Companion$readDexFiles$1$4.invoke(DexParser.kt:23)
	at kotlin.sequences.TransformingSequence$iterator$1.next(Sequences.kt:147)
	at kotlin.sequences.SequencesKt___SequencesKt.toCollection(_Sequences.kt:630)
	at kotlin.sequences.SequencesKt___SequencesKt.toMutableList(_Sequences.kt:660)
	at kotlin.sequences.SequencesKt___SequencesKt.toList(_Sequences.kt:651)
	at com.linkedin.dex.parser.DexParser$Companion.readDexFiles(DexParser.kt:80)
	at com.linkedin.dex.parser.DexParser$Companion.findTestMethods(DexParser.kt:58)
	at com.linkedin.dex.parser.DexParser.findTestMethods(DexParser.kt)
	at com.walmart.otto.Flank.getTestCaseNames(Flank.java:268)
	at com.walmart.otto.Flank.start(Flank.java:71)
	at com.walmart.otto.Flank.main(Flank.java:118)
```

As an upstream user of this project, thanks for all your hard work! By the way I posted the Flank issue here https://github.com/TestArmada/flank/issues/165

**Comment by drewhannay** [comment]
Thanks for the report, this is useful! I probably won't have time to take a look at this until after Google I/O next week, but I'll follow up.

**Comment by rdiaz82** [comment]
Confirmed! I got the same error using minSDK greater than 24. In order to check I changed to minSDK 18 and it works smoothly. Any idea?

**Comment by rdiaz82** [comment]
Hi, I have checked the code and I think that I found the bug source. The problem is caused because in android 7 the magic dex version has changed from 035 to 037 and in Android 8 it changes again from 037 to 038. here is further information about it https://source.android.com/devices/tech/dalvik/dex-format#dex-file-magic I have forked the repo and I going to try just to change the version to see if it works.
